### PR TITLE
Don't schedule on spot instances

### PR DESCRIPTION
**Pull request by dctrwatson**

### CODE REVIEW

**Comment by dctrwatson** [comment]
I assume this works even when the label doesn't exist at all....

File: cilium-etcd-cluster/descriptors.go [hunk 1]
```diff
@@ -61,6 +61,17 @@ func CiliumEtcdCluster(namespace, repository, version string, size int, etcdEnv
 					},
 				},
 				Affinity: &v1.Affinity{
+					NodeAffinity: &v1.NodeAffinity{
+						RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+							NodeSelectorTerms: []v1.NodeSelectorTerm{
+								MatchExpressions: []v1.NodeSelectorRequirement{
```
Comment (enisoc) [inline cilium-etcd-cluster/descriptors.go]: Should we go ahead and add `cloud.google.com/gke-preemptible != "true"` as well?

File: cilium-etcd-cluster/descriptors.go [hunk 2]
```diff
@@ -61,6 +61,17 @@ func CiliumEtcdCluster(namespace, repository, version string, size int, etcdEnv
...
+									Key:      "aws.amazon.com/lifecycle",
```
Comment (enisoc) [inline cilium-etcd-cluster/descriptors.go]: Where did you find this label? I can't find it documented anywhere.

**Review by enisoc — APPROVED** [review]
LGTM with optional suggestion.

> I assume this works even when the label doesn't exist at all....

I know this works as expected for Pod label selectors. I'd be surprised if the semantics were different for Node label selectors.

File: cilium-etcd-cluster/descriptors.go (hunk 2 above)
Comment (dctrwatson) [reply cilium-etcd-cluster/descriptors.go]: I'm adding it https://github.com/planetscale/planetscale-operator/pull/5140

File: cilium-etcd-cluster/descriptors.go (hunk 1 above)
Comment (dctrwatson) [reply cilium-etcd-cluster/descriptors.go]: ohhh, yes!

File: cilium-etcd-cluster/descriptors.go (hunk 2 above)
Comment (enisoc) [reply cilium-etcd-cluster/descriptors.go]: Ohh of course. I forgot that EKS doesn't manage workers for us.

**Review by enisoc — APPROVED** [review]
LGTM

**Pull request merged**